### PR TITLE
Pagination styling

### LINKED ISSUE #67: Pagination Styling
The search page has pagination, but it's default styles right now.

### PR DESCRIPTION
**Pull request by hannako**
closes #67

### CODE REVIEW

**Comment by hannako** [comment]
[Style pagination](https://trello.com/c/uwd13Iw2/343-style-pagination)

File: app/views/kaminari/_paginator.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    total_pages:   total number of pages
+    per_page:      number of items to fetch per page
+    remote:        data-remote
+    paginator:     the paginator that renders the pagination tags inside
+-%>
+
+<%= link_to_prev_page @datasets, 'Prev' unless current_page.first?%>
+
+<%= paginator.render do -%>
+  &nbsp
```
Comment (govuklaurence) [inline app/views/kaminari/_paginator.html.erb]: Why are these needed? Generally CSS should control spacing in cases like this.

File: config/initializers/kaminari_config.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+Kaminari.configure do |config|
+  config.default_per_page = 20
+  # config.max_per_page = nil
+  # config.window = 4
+  # config.outer_window = 0
+  # config.left = 0
+  # config.right = 0
+  # config.page_method_name = :page
+  # config.param_name = :page
+  # config.params_on_first_page = false
```
Comment (govuklaurence) [inline config/initializers/kaminari_config.rb]: Remove the commented bits

File: app/assets/stylesheets/components.scss [hunk 3]
```diff
@@ -194,6 +194,22 @@ section .year-expand {
   }
 }
 
+// Search results page _______________________________
+
+// Pagination
+.paginate{
```
Comment (maxf) [inline app/assets/stylesheets/components.scss]: Note: we should start doing BEM, otherwise it'll be a lot of work to convert everything later.

File: app/views/kaminari/_paginator.html.erb (hunk 1 above)
Comment (maxf) [inline app/views/kaminari/_paginator.html.erb]: should be `&nbsp;`

File: app/views/kaminari/_paginator.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  <% each_page do |page| -%>
+    <% if page.display_tag? -%>
+      <%= page_tag page %>
+    <% elsif !page.was_truncated? -%>
+      <%= gap_tag %>
+    <% end -%>
+    &nbsp
```
Comment (maxf) [inline app/views/kaminari/_paginator.html.erb]: ditto

File: app/views/kaminari/_paginator.html.erb (hunk 1 above)
Comment (hannako) [inline app/views/kaminari/_paginator.html.erb]: Yes, I was being lazy/following how the kaminari views did it. I'll fix

File: config/initializers/kaminari_config.rb (hunk 2 above)
Comment (hannako) [inline config/initializers/kaminari_config.rb]: 👍

File: config/locales/en.yml [hunk 5]
```diff
@@ -30,4 +30,19 @@
...
+      last: "Last &raquo;"
+      previous: "&lsaquo; Prev"
+      next: "Next &rsaquo;"
+      truncate: "&hellip;"
+  helpers:
+    page_entries_info:
+      one_page:
+        display_entries:
+          zero: "No %{entry_name} found"
+          one: "Showing <b>1</b> %{entry_name}"
```
Comment (maxf) [inline config/locales/en.yml]: Are these used? If not, you should remove

File: app/views/search/search.html.erb [hunk 6]
```diff
@@ -60,7 +60,12 @@
             <p><%= truncate(strip_tags(d.summary), length: 200, separator: ' ') %></p>
           </div>
         <% end %>
-        <%= paginate @datasets %>
+        <div class="pagination">
+          <nav>
+            <%= page_entries_info @datasets, entry_name: 'dataset' %>
+            <span class="paginate"><%= paginate @datasets, class:'paginate' %></span>
+          </nav>
```
Comment (maxf) [inline app/views/search/search.html.erb]: For accessibility reasons, it would be good to have invisible text for screen reader, in front of each link number, eg `<a href="..."><span class="visuallyhidden">Page </span>1</a>`. Is it possible with Kaminari?

File: config/locales/en.yml (hunk 5 above)
Comment (hannako) [inline config/locales/en.yml]: the `one:` is used, but the `zero:` is currently handled by your zero search results PR. I guess I thought it was belt and braces, but I will remove it.
